src/test/basics/PerfLog_test.cpp

```diff
@@ -214,6 +219,38 @@ class PerfLog_test : public beast::unit_test::suite
         return currents;
     }
 
+//------------------------------------------------------------------------------
+
+    // Helper function that checks the size of the PerfLog file and then
+    // returns when the file gets bigger.  This indicates that the PerfLog
+    // has written new values to the file and _should_ have the latest
+    // update.
+    static void waitForFileUpdate (PerfLogParent const& parent)
+    {
+        using namespace boost::filesystem;
+
+        auto const path = parent.getPerfLogPath() / parent.getPerfLogFileName();
+        if (!exists (path))
+            return;
+
+        // We wait for the file to change size twice.  The first file size
+        // change may have been in process while we arrived.
+        std::uintmax_t const firstSize {file_size (path)};
```

Comment:
I have very slight preference for `auto` in this case since all the values being compared are `decltype<file_size()>`, but I'm definitely fine with the explicit types too.